query/Undefiner.java

```diff
@@ -167,14 +167,13 @@ private void undefineSub(ThingType thingType, SubConstraint subConstraint) {
             final ThingType supertype = getThingType(subConstraint.type().label().get());
             if (supertype == null) {
                 throw GraknException.of(TYPE_NOT_FOUND, subConstraint.type().label().get());
-            } else if (thingType.getSupertypes().noneMatch(t -> t.equals(supertype))
-                    && !(supertype instanceof ThingTypeImpl.Root)) {
+            } else if (thingType.getSupertypes().noneMatch(t -> t.equals(supertype))) {
                 throw GraknException.of(INVALID_UNDEFINE_SUB, thingType.getLabel(), supertype.getLabel());
             }
             if (thingType instanceof RelationType) {
                 variables.stream().filter(
                         v -> v.label().isPresent() && v.label().get().scope().isPresent() &&
-                                v.label().get().scope().get().equals(thingType.getLabel().scope().get())
+                                v.label().get().scope().get().equals(thingType.getLabel().toString())
```

Comment:
This was causing a rocksdb error- we are asking if the scope of the roletype is the same as the type of the relation we are looking at- that is, if the role is scoped to this relation. Relations do not themselves have scope, hence this change.

Reply:
why call `toString()` here? that's a bit weird. `toString()` doesn't describe the property of the object, but rather a string representation of the object. This should return `.name()` ?